.github/workflows/provider.yml

```diff
@@ -0,0 +1,133 @@
+        working-directory: ${{inputs.name}}
+
+      - name: Store artifact
+        uses: actions/upload-artifact@v4
+        #if: startswith(github.ref, 'refs/tags/v') # Only run on tag push
+        with:
+          name: ${{inputs.name}}-${{matrix.arch}}
+          path: ${{inputs.name}}/${{inputs.name}}
+  release:
+    needs: [build]
+    #if: startsWith(github.ref, 'refs/tags/${{inputs.name}}-v')
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Extract tag context
+        id: ctx
+        run: |
+          sha_short=$(git rev-parse --short HEAD)
+          echo "owner=${GITHUB_REPOSITORY_OWNER,,}" >> "$GITHUB_OUTPUT"
+          echo "sha_short=${sha_short}" >> "$GITHUB_OUTPUT"
+          echo "version=${GITHUB_REF#refs/tags/${{inputs.name}}v}" >> "$GITHUB_OUTPUT"
+          echo "artifact_name=$(echo ${{inputs.name}} | tr '- "')" >> "$GITHUB_OUTPUT"
+      - uses: taiki-e/install-action@v2
+        with:
+          tool: wash-cli@0.28.0
+
+      # Used later in the build process to push provenance info
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v3
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      #- uses: imjasonh/setup-crane@v0.1
+
+      - uses: actions/download-artifact@v4
+        with:
+          path: artifacts
+
+      - name: Fix permissions and architectures
+        run: |
+          tree artifacts
+          mv artifacts/${{inputs.name}}-x86_64/${{inputs.name}} ${{inputs.name}}-x86_64
+          mv artifacts/${{inputs.name}}-aarch64/${{inputs.name}} ${{inputs.name}}-aarch64
+          chmod +x ${{inputs.name}}-*
+
+      - name: Push
+        env:
+          WASH_REG_USER: ${{ github.repository_owner }}
+          WASH_REG_PASSWORD: ${{ secrets.GITHUB_TOKEN }}
+          WASH_ISSUER_KEY: ${{ secrets.subject}}
+          WASH_SUBJECT_KEY: ${{ secrets.issuer}}
+        run: |
+          if [ -z $WASH_SUBJECT_KEY ]; then
+            echo "WASH_SUBJECT_KEY is required"
+            exit 1
+          fi
+
+          if [ -z $WASH_ISSUER_KEY ]; then
+            echo "WASH_ISSUER_KEY is required"
+            exit 1
+          fi
+
+          wash par create \
+          --binary "${{ inputs.name }}-x86_64" \
+          --compress \
+          --destination "${{ inputs.name }}.par.gz" \
+          --name "${{ inputs.name }}-provider" \
+          --vendor cosmonic-labs \
+          --version ${{ steps.ctx.outputs.version }}
+
+          wash par insert --arch aarch64-linux --binary "${{inputs.name }}-aarch64" "${{ inputs.name }}.par.gz"
+
+          wash push ${{env.REGISTRY}}/${{env.REPOSITORY}}/${{inputs.name}}:$GITHUB_SHA ${{inputs.name}}.par.gz
+          wash push ${{env.REGISTRY}}/${{env.REPOSITORY}}/${{inputs.name}}:$(git rev-parse --short HEAD) ${{inputs.name}}.par.gz
```

Comment:
This is admittedly nitpicky, but if you wanted to avoid having to fetch the short SHA to match GitHub's style, you could also use [substring expansion with `${GITHUB_SHA::7}`](https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html) here